### ISSUE TITLE
implementation of the REST API for the event feature

### ISSUE THREAD

**Issue by adekoder** [pyung/pyung.github.io#14]

**Comment by adekoder** [comment]
i created this issue so that we can discuss how we plan on implementing this feature , also to decide the kind of data we will be collecting

**Comment by adekoder** [comment]
@wolejnr i need you here

**Comment by gbozee** [comment]
great. let me add the other assignees at the side.

**Comment by gbozee** [comment]
take a look at https://djangogirls.org/events/ to get an idea of what you would be working on.

**Comment by adekoder** [comment]
That i have done

**Comment by adekoder** [comment]
we just need to finalize on the set of data we will been rendering
eg : event date , location , time etc

**Comment by wolejnr** [comment]
I am with you @adekoder. 
As regard implementation I guess we would be using gatsbyJS. Are you familiar with this?
As regards the data to be collected, we can try to spot the common things in an event such as the ones you mentioned.
> eg : event date , location , time etc

I think we can split location into `venue` and `state` or what do you think? 
I also think we will be needing to render About, Apply, FAQ, Be a Mentor and Partners

**Comment by adekoder** [comment]
"gatsbyJS" i have no idea what it is but will do my research,
splitting the venue and state is nice
i don't know why we will need "Be a mentor"

**Comment by adekoder** [comment]
i thought we are only implementing the API why do we need  this "gatsbyJS"

**Comment by gbozee** [comment]
yeah its the Api you would be concerned about. That would be done in django

**Comment by adekoder** [comment]
then why do we need "gatsbyJS"

**Comment by gbozee** [comment]
You are working on APIs not the frontend. It is the frontend team that needs to learn gatsbyjs. Not all part of the site would be built with django. In fact django would just be exposed as a rest Api only. We are not using the templates. Gatsby is a static generator site that ensures we are just deploying HTML and CSS and JavaScript in d end, no server to host. 

The whole django setup is useful from an admin point of view not from a user/frontend view.
You could think about it that way.

**Comment by wolejnr** [comment]
Thanks for the clarification @gbozee

**Comment by wolejnr** [comment]
@adekoder based on your comment:
> i don't know why we will need "Be a mentor"

Just a thought especially since we are trying to pattern after DjangoGirls.

**Comment by adekoder** [comment]
ok, we can add it if we want, but i don't see any reason to

**Comment by wolejnr** [comment]
So my suggestions as regards the data to be rendered for events are given as follows:
- [ ] Event start date
- [ ] Event end date
- [ ] Venue
- [ ] State (e.g. Kaduna, Kebbi etc.)
- [ ] Event URL
- [ ] Sponsors
- [ ] FAQ

Check those that you agree with and include those you think should also be there.

**Comment by wolejnr** [comment]
@adekoder Do you agree with the suggestions for data to be rendered for events??

**Comment by adekoder** [comment]
sure

**Comment by gbozee** [comment]
Based on the list you gave @wolejnr There are some fields i am not sure would be required yet. Try reaching out to @shante66 for more information on what to consider here
